reactor-core/src/main/java/reactor/core/publisher/SerializedSubscriber.java

```diff
@@ -148,6 +150,11 @@ public void cancel() {
 	}
 
 	void serAdd(T value) {
+		if (cancelled) {
+			Operators.onDiscard(value, actual.currentContext());
```

Comment:
same as above

Reply:
here I don't think so, since we're dealing with cancellation inside `onNext` (which is precisely what `doOnDiscard` is about)